props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+      }
+    } else {
+      if (!hybrid_font.PlayPolyphonic(&SFX_bmend)) {
+        beeper.Beep(0.5, 3000);
+      }
+    }
 #endif
-        VolumeMenu();
-        return true;
+  }
+
+#ifdef BC_DUAL_BLADES
+
+  static const BladeSet controlled_blades_;
+
+  bool isMainBladeOn() { return SaberBase::OnBlades()[BC_MAIN_BLADE]; }
+  bool isSecondBladeOn() { return SaberBase::OnBlades()[BC_SECOND_BLADE]; }
+
+  // Get blade location to apply effect to
+  BladeSet GetBladeAboveHorizon() {
+    return accel_.x > 0 ? BC_MAIN_BLADE_SET : BC_SECOND_BLADE_SET;
+  }
+
+  void TurnBladeOn(BladeSet target_blade) {
+    // Add in all non-controlled blades, effectively excluding the "other" blade.
+    target_blade = target_blade | ~controlled_blades_;
+    if (SaberBase::OnBlades().off()) {
+      PVLOG_NORMAL << "** No blades are currently ON, turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade and all others, excluding the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "SECOND" : "MAIN")
+                   << " blade\n";
+      if (isPointingUp() || muted_) {
+        FastOn(EffectLocation(0, target_blade));
+     } else {
+        On(EffectLocation(0, target_blade));
+      }
+    } else {
+      PVLOG_NORMAL << "** Turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade\n";
+      SaberBase::TurnOn(EffectLocation(0, target_blade));
+    }
+  }
+
+  void TurnBladeOff(BladeSet target_blade) {
+    SaberBase::OffType off_type = isPointingUp() ? SaberBase::OFF_FAST : SaberBase::OFF_NORMAL;
+    PVLOG_DEBUG << "***** Off type: " << (off_type == SaberBase::OFF_FAST ? "OFF_FAST" : "OFF_NORMAL") << "\n";
+
+    // Check if this is the only blade ON (of MAIN or SECOND blades)
+    if ((SaberBase::OnBlades() & ~target_blade & controlled_blades_).off()) {
+      // Other blade is not on, so just do normal Off() with appropriate off_type
+      PVLOG_NORMAL << "** Turning OFF all blades\n";
+      Off(off_type);
+      muted_ = false;
+    } else {
+      // Only Turn OFF this blade, leave the other one ON.
+      PVLOG_NORMAL << "** Turning OFF only the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND") 
+                   << " blade\n";
+      SaberBase::TurnOff(off_type, EffectLocation(0, target_blade));
+    }
+    battle_mode_ = false;
+    spam_blast_ = false;
+    saber_off_time_ = millis();
+  }
+
+  void TurnMainBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_mainBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      mute_mainBlade_timer_.trigger(0);
+    }
+  }
+
+  void TurnSecondBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_secondBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      mute_mainBlade_timer_.trigger(0);
+    }
+  }
+
+  void TurnDualBladesOn() {
```

Comment:
This function (and the next one) seem kind of pointless...

Reply:
I guess that was from before TurnOn/OffHelper() existed, and then I left them for printouts for testing.
Adjusted.